internal/resources/google/container_node_pool.go

```diff
@@ -45,10 +47,15 @@ func (r *ContainerNodePool) BuildResource() *schema.Resource {
 	poolSize := int64(1)
 
 	costComponents := []*schema.CostComponent{
-		computeCostComponent(r.Region, r.NodeConfig.MachineType, r.NodeConfig.PurchaseOption, poolSize, nil),
 		computeDiskCostComponent(r.Region, r.NodeConfig.DiskType, r.NodeConfig.DiskSize, poolSize),
 	}
 
+	if strings.Contains(r.NodeConfig.MachineType, "custom") {
+		costComponents = append(costComponents, customComputeCostComponents(r.Region, r.NodeConfig.MachineType, r.NodeConfig.PurchaseOption, poolSize, nil)...)
+	} else {
+		costComponents = append(costComponents, computeCostComponent(r.Region, r.NodeConfig.MachineType, r.NodeConfig.PurchaseOption, poolSize, nil))
+	}
```

Comment:
One idea would be just to create a `computeCostComponents` function and we could put this condition in there. We can then update every google resource that uses `computeCostComponent` to use the new function instead.

That way we don't need to check the machine type for each resource that uses compute components. For example we also have `google_compute_instance_group` and `google_compute_region_instance_group` that use the `computeCostComponent` method.

Reply:
Interesting. I just fixed the queries in the latest commits. I'll think about this soon.

Reply:
👍 awesome, thanks!

Reply:
wdyt https://github.com/infracost/infracost/pull/2004/commits/f24e61e6e761422ef11e115c40c079f8e831d25b?

Reply:
@aliscott throw me another thorough code review, please o/